pvlib/modelchain.py

```diff
@@ -961,6 +1035,97 @@ def run_model(self, weather, times=None):
         self.aoi_model()
         self.spectral_model()
         self.effective_irradiance_model()
+
+        self.run_model_from_effective_irradiance()
+
+        return self
+
+    def run_model_from_poa(self, total_irrad):
```

Comment:
To keep the API consistent with `run_model`, I think we should keep the full `weather` DataFrame as the input here and the method below.

Reply:
As an input in addition to `total_irrad`? Or assume that `poa_global` etc are in `weather`? That would be different from what `ModelChain.prepare_inputs` does currently (input is `weather`, output is `total_irrad` with the POA components)